code/modules/mob/mob_movement.dm

```diff
@@ -302,7 +302,7 @@
 
 		//We are now going to move
 		move_delay = max(move_delay,1)
-		if(mob.movement_speed_modifier)
+		if(mob.movement_speed_modifier && mob.movement_speed_modifier != 1)
```

Comment:
You don't need to change this, it doesn't runtime when it's 1, it only does when it's 0 or null due to the division in the next line.